mesonbuild/compilers/c.py

```diff
@@ -692,10 +693,11 @@ def has_function(self, funcname, prefix, env, extra_args=None, dependencies=None
         # need to look for them differently. On nice compilers like clang, we
         # can just directly use the __has_builtin() macro.
         fargs['no_includes'] = '#include' not in prefix
+        fargs['func_is_builtin'] = funcname.startswith('__builtin_')
         t = '''{prefix}
         int main() {{
         #ifdef __has_builtin
-            #if !__has_builtin(__builtin_{func})
+            #if !__has_builtin({func})
```

Comment:
:+1:

But does this work without also adding implicit includes for the cases where users currently got away with not giving the includes?

Reply:
Yes, the alloca test checks for this already (on clang).

Reply:
Yes, it seems to work with `alloca`, but with `isgreaterequal` it does not. Now it might be my clang install (it's just for compiling qtcreator). But a simple manual build test with clang seems to work. But even `has_function('isgreaterequal', prefix: '#include <math.h>')` does not work for me at the moment. It works with a random older meson.